<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>QC and downstream analysis for differential expression RNA-seq • DEGreport</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="QC and downstream analysis for differential expression RNA-seq">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEGreport</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.17.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEGreport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>QC and downstream analysis for differential expression RNA-seq</h1>
                        <h4 class="author">Lorena Pantano</h4>
            
            <h4 class="date">12 July 2018</h4>
      
      
      <div class="hidden name"><code>DEGreport.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>DEGreport package version: 1.17.3</p>
    </div>
    
<p><a href="lorena.pantano@gmail.com">Lorena Pantano</a> Harvard TH Chan School of Public Health, Boston, US</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DEGreport)
<span class="kw">data</span>(humanGender)</code></pre></div>
<div id="general-qc-figures-from-de-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#general-qc-figures-from-de-analysis" class="anchor"></a>General QC figures from DE analysis</h2>
<p>We are going to do a differential expression analysis with edgeR/DESeq2. We have an object that is coming from the edgeR package. It contains a gene count matrix for 85 TSI HapMap individuals, and the gender information. With that, we are going to apply the <code>glmFit</code> function or <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em> to get genes differentially expressed between males and females.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DESeq2)
idx &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">75</span><span class="op">:</span><span class="dv">85</span>)
dds &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/DESeqDataSet">DESeqDataSetFromMatrix</a></span>(<span class="kw">assays</span>(humanGender)[[<span class="dv">1</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>, idx],
                              <span class="kw">colData</span>(humanGender)[idx,], <span class="dt">design=</span><span class="op">~</span>group)
dds &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/DESeq">DESeq</a></span>(dds)
res &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/results">results</a></span>(dds)</code></pre></div>
<p>We need to extract the experiment design data.frame where the condition is Male or Female.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds, <span class="dt">normalized =</span> <span class="ot">TRUE</span>)
design &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(dds))</code></pre></div>
<div id="size-factor-qc" class="section level3">
<h3 class="hasAnchor">
<a href="#size-factor-qc" class="anchor"></a>Size factor QC</h3>
<p>A main assumption in library size factor calculation of edgeR and DESeq2 (and others) is that the majority of genes remain unchanged. Plotting the distribution of gene ratios between each gene and the average gene can show how true this is. Not super useful for many samples because the plot becomes crowed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degCheckFactors.html">degCheckFactors</a></span>(counts[, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>])</code></pre></div>
<p><img src="DEGreport_files/figure-html/check-factor-1.png" width="700"></p>
</div>
<div id="mean-variance-qc-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#mean-variance-qc-plots" class="anchor"></a>Mean-Variance QC plots</h3>
<p>p-value distribution gives an idea on how well you model is capturing the input data and as well whether it could be some problem for some set of genes. In general, you expect to have a flat distribution with peaks at 0 and 1. In this case, we add the mean count information to check if any set of genes are enriched in any specific p-value range.</p>
<p>Variation (dispersion) and average expression relationship shouldn’t be a factor among the differentially expressed genes. When plotting average mean and standard deviation, significant genes should be randomly distributed.</p>
<p>In this case, it would be good to look at the ones that are totally outside the expected correlation.</p>
<p>You can put this tree plots together using <code>degQC</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degQC.html">degQC</a></span>(counts, design[[<span class="st">"group"</span>]], <span class="dt">pvalue =</span> res[[<span class="st">"pvalue"</span>]])</code></pre></div>
<p><img src="DEGreport_files/figure-html/qc-1.png" width="700"></p>
</div>
<div id="covariates-effect-on-count-data" class="section level3">
<h3 class="hasAnchor">
<a href="#covariates-effect-on-count-data" class="anchor"></a>Covariates effect on count data</h3>
<p>Another important analysis to do if you have covariates is to calculate the correlation between PCs from PCA analysis to different variables you may think are affecting the gene expression. This is a toy example of how the function works with raw data, where clearly library size correlates with some of the PCs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resCov &lt;-<span class="st"> </span><span class="kw"><a href="../reference/degCovariates.html">degCovariates</a></span>(<span class="kw">log2</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds)<span class="op">+</span><span class="fl">0.5</span>),
                        <span class="kw">colData</span>(dds))</code></pre></div>
<p><img src="DEGreport_files/figure-html/cov-1.png" width="700"></p>
</div>
<div id="covariates-correlation-with-metrics" class="section level3">
<h3 class="hasAnchor">
<a href="#covariates-correlation-with-metrics" class="anchor"></a>Covariates correlation with metrics</h3>
<p>Also, the correlation among covariates and metrics from the analysis can be tested. This is useful when the study has multiple variables, like in clinical trials. The following code will return a correlation table, and plot the correlation heatmap for all the covariates and metrics in a table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cor &lt;-<span class="st"> </span><span class="kw"><a href="../reference/degCorCov.html">degCorCov</a></span>(<span class="kw">colData</span>(dds))</code></pre></div>
<p><img src="DEGreport_files/figure-html/corcov-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(cor)</code></pre></div>
<pre><code>## [1] "cor"    "corMat" "fdrMat" "plot"</code></pre>
</div>
<div id="qc-report" class="section level3">
<h3 class="hasAnchor">
<a href="#qc-report" class="anchor"></a>QC report</h3>
<p>A quick HTML report can be created with <code>createReport</code> to show whether a DE analysis is biased to a particular set of genes. It contains the output of <code>degQC,</code>degVB and <code>degMB</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/createReport.html">createReport</a></span>(<span class="kw">colData</span>(dds)[[<span class="st">"group"</span>]], <span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds, <span class="dt">normalized =</span> <span class="ot">TRUE</span>),
             <span class="kw">row.names</span>(res)[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>], res[[<span class="st">"pvalue"</span>]], <span class="dt">path =</span> <span class="st">"~/Downloads"</span>)</code></pre></div>
</div>
</div>
<div id="report-from-deseq2-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#report-from-deseq2-analysis" class="anchor"></a>Report from DESeq2 analysis</h2>
<p>Here, we show some useful plots for differentially expressed genes.</p>
<div id="contrasts" class="section level3">
<h3 class="hasAnchor">
<a href="#contrasts" class="anchor"></a>Contrasts</h3>
<p><code>DEGSet</code> is a class to store the DE results like the one from <code>results</code> function. <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em> offers multiple way to ask for contrasts/coefficients. With <code>degComps</code> is easy to get multiple results in a single object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">degs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/degComps.html">degComps</a></span>(dds, <span class="dt">combs =</span> <span class="st">"group"</span>,
                 <span class="dt">contrast =</span> <span class="kw">list</span>(<span class="st">"group_Male_vs_Female"</span>,
                                 <span class="kw">c</span>(<span class="st">"group"</span>, <span class="st">"Female"</span>, <span class="st">"Male"</span>)))
<span class="kw">names</span>(degs)</code></pre></div>
<pre><code>## [1] "group_Male_vs_Female" "group_Female_vs_Male"</code></pre>
<p><code>degs</code> contains 3 elements, one for each contrast/coefficient asked for. It contains the results output in the element <code>raw</code> and the output of <code>lfcShrink</code> in the element <code>shrunken</code>. To obtain the results from one of them, use the method <code>dge</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/deg.html">deg</a></span>(degs[[<span class="dv">1</span>]])</code></pre></div>
<pre><code>## log2 fold change (MAP): group Male vs Female 
## Wald test p-value: group Male vs Female 
## DataFrame with 1000 rows and 6 columns
##                   baseMean log2FoldChange      lfcSE         stat
##                  &lt;numeric&gt;      &lt;numeric&gt;  &lt;numeric&gt;    &lt;numeric&gt;
## ENSG00000067048 1025.03783      1.9394875 0.10069402    23.994380
## ENSG00000012817  411.54387      3.7005640 0.09898885    21.804517
## ENSG00000067646  169.81477      3.3228457 0.09951933    15.483847
## ENSG00000005889  670.86191     -0.4894347 0.09293631    -5.263708
## ENSG00000006757   92.66111     -0.4729262 0.09927492    -4.757006
## ...                    ...            ...        ...          ...
## ENSG00000068120  1214.0967  -0.0001143219 0.07972266 -0.001433998
## ENSG00000072062   935.3172   0.0005592191 0.09153522  0.006108720
## ENSG00000076770  1019.7964   0.0008500818 0.10166633  0.008361930
## ENSG00000078967   166.4221   0.0004157454 0.09597580  0.004329710
## ENSG00000079246  5226.3390   0.0001495843 0.09262051  0.001614995
##                        pvalue          padj
##                     &lt;numeric&gt;     &lt;numeric&gt;
## ENSG00000067048 3.183020e-127 3.183020e-124
## ENSG00000012817 2.102139e-105 1.051070e-102
## ENSG00000067646  4.459844e-54  1.486615e-51
## ENSG00000005889  1.411785e-07  3.529463e-05
## ENSG00000006757  1.964856e-06  3.929713e-04
## ...                       ...           ...
## ENSG00000068120     0.9988558     0.9988558
## ENSG00000072062     0.9951260     0.9988558
## ENSG00000076770     0.9933282     0.9988558
## ENSG00000078967     0.9965454     0.9988558
## ENSG00000079246     0.9987114     0.9988558</code></pre>
<p>By default it would output the <code>shrunken</code> table always, as defined by <code>degDefault</code>, that contains the default table to get.</p>
<p>To get the original results table, use the parameter as this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/deg.html">deg</a></span>(degs[[<span class="dv">1</span>]], <span class="st">"raw"</span>, <span class="st">"tibble"</span>)</code></pre></div>
<pre><code>## # A tibble: 1,000 x 7
##    gene          baseMean log2FoldChange  lfcSE   stat    pvalue      padj
##  * &lt;chr&gt;            &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##  1 ENSG00000067…   1025.          10.2   0.423   24.0  3.18e-127 3.18e-124
##  2 ENSG00000012…    412.           9.24  0.424   21.8  2.10e-105 1.05e-102
##  3 ENSG00000067…    170.          10.2   0.658   15.5  4.46e- 54 1.49e- 51
##  4 ENSG00000005…    671.          -0.692 0.131   -5.26 1.41e-  7 3.53e-  5
##  5 ENSG00000006…     92.7         -0.767 0.161   -4.76 1.96e-  6 3.93e-  4
##  6 ENSG00000073…    220.          -1.87  0.421   -4.44 8.89e-  6 1.48e-  3
##  7 ENSG00000005…   2027.          -0.742 0.176   -4.21 2.59e-  5 3.69e-  3
##  8 ENSG00000005…   1234.           0.389 0.0952   4.08 4.44e-  5 5.56e-  3
##  9 ENSG00000003…    394.           0.680 0.177    3.85 1.17e-  4 1.31e-  2
## 10 ENSG00000069…    107.          -1.63  0.459   -3.56 3.71e-  4 3.71e-  2
## # ... with 990 more rows</code></pre>
<p>Note that the format of the output can be changed to tibble, or data.frame with a third parameter <code>tidy</code>.</p>
<p>The table will be always sorted by padj.</p>
<p>And easy way to get significant genes is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/significants.html">significants</a></span>(degs[[<span class="dv">1</span>]], <span class="dt">fc =</span> <span class="dv">0</span>, <span class="dt">fdr =</span> <span class="fl">0.05</span>)</code></pre></div>
<pre><code>##  [1] "ENSG00000012817" "ENSG00000067646" "ENSG00000067048"
##  [4] "ENSG00000005889" "ENSG00000006757" "ENSG00000005302"
##  [7] "ENSG00000003400" "ENSG00000073282" "ENSG00000005020"
## [10] "ENSG00000069702"</code></pre>
<p>This function can be used as well for a list of comparisons:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/significants.html">significants</a></span>(degs, <span class="dt">fc =</span> <span class="dv">0</span>, <span class="dt">fdr =</span> <span class="fl">0.05</span>)</code></pre></div>
<pre><code>##  [1] "ENSG00000012817" "ENSG00000067646" "ENSG00000067048"
##  [4] "ENSG00000005889" "ENSG00000006757" "ENSG00000005302"
##  [7] "ENSG00000003400" "ENSG00000073282" "ENSG00000005020"
## [10] "ENSG00000069702"</code></pre>
<p>And it can returns the full table for a list:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/significants.html">significants</a></span>(degs, <span class="dt">fc =</span> <span class="dv">0</span>, <span class="dt">fdr =</span> <span class="fl">0.05</span>, <span class="dt">full =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## # A tibble: 10 x 5
##    gene     log2FoldChange_group… log2FoldChange_group… padj_group_Female…
##    &lt;chr&gt;                    &lt;dbl&gt;                 &lt;dbl&gt;              &lt;dbl&gt;
##  1 ENSG000…                -0.494                 0.388          1.31e-  2
##  2 ENSG000…                -0.351                 0.319          5.56e-  3
##  3 ENSG000…                 0.541                -0.425          3.69e-  3
##  4 ENSG000…                 0.573                -0.489          3.53e-  5
##  5 ENSG000…                 0.585                -0.473          3.93e-  4
##  6 ENSG000…                -5.01                  3.70           1.05e-102
##  7 ENSG000…                -3.63                  1.94           3.18e-124
##  8 ENSG000…                -4.43                  3.32           1.49e- 51
##  9 ENSG000…                 0.453                -0.260          3.71e-  2
## 10 ENSG000…                 0.582                -0.339          1.48e-  3
## # ... with 1 more variable: padj_group_Male_vs_Female &lt;dbl&gt;</code></pre>
<p>Since log2FoldChange are shrunken, the method for DEGSet class now can plot these changes as follow:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degMA.html">degMA</a></span>(degs[[<span class="dv">1</span>]], <span class="dt">diff =</span> <span class="dv">2</span>, <span class="dt">limit =</span> <span class="dv">3</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/plotMA-1.png" width="700"></p>
<p>The blue arrows indicate how foldchange is affected by this new feature.</p>
<p>As well, it can plot the original MA plot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degMA.html">degMA</a></span>(degs[[<span class="dv">1</span>]], <span class="dt">diff =</span> <span class="dv">2</span>, <span class="dt">limit =</span> <span class="dv">3</span>, <span class="dt">raw =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/plotMA-raw-1.png" width="700"></p>
<p>or the correlation between the original log2FoldChange and the new ones:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degMA.html">degMA</a></span>(degs[[<span class="dv">1</span>]], <span class="dt">limit =</span> <span class="dv">3</span>, <span class="dt">correlation =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/plotMA-cor-1.png" width="700"></p>
</div>
<div id="volcano-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h3>
<p>Volcano plot using the output of <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em>. It mainly needs data.frame with two columns (logFC and pVal). Specific genes can be plot using the option <code>plot\_text</code> (subset of the previous data.frame with a 3rd column to be used to plot the gene name).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res[[<span class="st">"id"</span>]] &lt;-<span class="st"> </span><span class="kw">row.names</span>(res)
show &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(res[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">"log2FoldChange"</span>, <span class="st">"padj"</span>, <span class="st">"id"</span>)])
<span class="kw"><a href="../reference/degVolcano.html">degVolcano</a></span>(res[,<span class="kw">c</span>(<span class="st">"log2FoldChange"</span>, <span class="st">"padj"</span>)], <span class="dt">plot_text =</span> show)</code></pre></div>
<p><img src="DEGreport_files/figure-html/deseq2-volcano-1.png" width="700"></p>
<p>Note that the function is compatible with DEGset. Using <code><a href="../reference/degVolcano.html">degVolcano(degs[[1]])</a></code> is valid.</p>
</div>
<div id="gene-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-plots" class="anchor"></a>Gene plots</h3>
<p>Plot top genes coloring by group. Very useful for experiments with nested groups. <code>xs</code> can be <code>time</code> or <code>WT</code>/<code>KO</code>, and <code>group</code> can be <code>treated</code>/<code>untreated</code>. Another classification can be added, like <code>batch</code> that will plot points with different shapes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degPlot.html">degPlot</a></span>(<span class="dt">dds =</span> dds, <span class="dt">res =</span> res, <span class="dt">n =</span> <span class="dv">6</span>, <span class="dt">xs =</span> <span class="st">"group"</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/deseq2-gene-plots-1.png" width="700"></p>
<p>Another option for plotting genes in a wide format:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degPlotWide.html">degPlotWide</a></span>(dds, <span class="kw">rownames</span>(dds)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>], <span class="dt">group=</span><span class="st">"group"</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/deseq2-gene-plot-wide-1.png" width="700"></p>
</div>
<div id="markers-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#markers-plots" class="anchor"></a>Markers plots</h3>
<p>Markers can be used to show whether different conditions are enriched in different markers. For instance, in this example, Females and Males show different total expression for chromosome X/Y markers</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(geneInfo)
<span class="kw"><a href="../reference/degSignature.html">degSignature</a></span>(humanGender, geneInfo, <span class="dt">group =</span> <span class="st">"group"</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/markers-1.png" width="700"></p>
</div>
<div id="full-report" class="section level3">
<h3 class="hasAnchor">
<a href="#full-report" class="anchor"></a>Full report</h3>
<p>If you have a DESeq2 object, you can use degResults to create a full report with markdown code inserted, including figures and table with top de-regulated genes, GO enrichment analysis and heatmaps and PCA plots. If you set , different files will be saved there.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resreport &lt;-<span class="st"> </span><span class="kw"><a href="../reference/degResults.html">degResults</a></span>(<span class="dt">dds =</span> dds, <span class="dt">name =</span> <span class="st">"test"</span>, <span class="dt">org =</span> <span class="ot">NULL</span>,
                        <span class="dt">do_go =</span> <span class="ot">FALSE</span>, <span class="dt">group =</span> <span class="st">"group"</span>, <span class="dt">xs =</span> <span class="st">"group"</span>,
                        <span class="dt">path_results =</span> <span class="ot">NULL</span>)</code></pre></div>
<pre><code>## ## Comparison:  test {.tabset} 
## 
## 
##  &lt;br&gt;out of 1000 with nonzero total read count&lt;br&gt;adjusted p-value &lt; 0.1&lt;br&gt;LFC &gt; 0 (up)     : 6, 0.6% &lt;br&gt;LFC &lt; 0 (down)   : 7, 0.7% &lt;br&gt;outliers [1]     : 0, 0% &lt;br&gt;low counts [2]   : 0, 0% &lt;br&gt;(mean count &lt; 47)&lt;br&gt;[1] see 'cooksCutoff' argument of ?results&lt;br&gt;[2] see 'independentFiltering' argument of ?results&lt;br&gt;&lt;br&gt;NULL 
## 
## 
## Differential expression file at:  test_de.csv
## 
## Normalized counts matrix file at:  test_log2_counts.csv
## 
## ### MA plot plot</code></pre>
<p><img src="DEGreport_files/figure-html/deseq2-1.png" width="700"></p>
<pre><code>## 
## 
## ### Volcano plot
## 
## 
## 
## ### QC for DE genes</code></pre>
<p><img src="DEGreport_files/figure-html/deseq2-2.png" width="700"></p>
<pre><code>## 
## 
## ### Most significants, FDR&lt; 0.05  and log2FC &gt;  0.1 :  10</code></pre>
<p><img src="DEGreport_files/figure-html/deseq2-3.png" width="700"></p>
<pre><code>## 
## 
## 
## ### Plots top 9 most significants
## 
## 
## 
## ### Top DE table
## 
## 
## 
##                      baseMean   log2FoldChange       lfcSE        stat      pvalue        padj   absMaxLog2FC
## ----------------  -----------  ---------------  ----------  ----------  ----------  ----------  -------------
## ENSG00000067048    1025.03783       10.1571705   0.4233146   23.994380   0.0000000   0.0000000     10.1571705
## ENSG00000012817     411.54387        9.2394007   0.4237379   21.804517   0.0000000   0.0000000      9.2394007
## ENSG00000067646     169.81477       10.1874916   0.6579432   15.483847   0.0000000   0.0000000     10.1874916
## ENSG00000005889     670.86191       -0.6919265   0.1314523   -5.263708   0.0000001   0.0000353      0.6919265
## ENSG00000006757      92.66111       -0.7666012   0.1611520   -4.757006   0.0000020   0.0003930      0.7666012
## ENSG00000073282     220.15603       -1.8685615   0.4206120   -4.442482   0.0000089   0.0014821      1.8685615
## ENSG00000005302    2026.54990       -0.7418952   0.1763412   -4.207157   0.0000259   0.0036943      0.7418952
## ENSG00000005020    1233.86316        0.3888370   0.0952312    4.083085   0.0000444   0.0055552      0.3888370
## ENSG00000003400     393.62677        0.6803243   0.1766475    3.851310   0.0001175   0.0130542      0.6803243
## ENSG00000069702     106.67010       -1.6323189   0.4585611   -3.559654   0.0003713   0.0371343      1.6323189
## ENSG00000010278      84.30823        1.2035871   0.3554857    3.385754   0.0007098   0.0645300      1.2035871
## ENSG00000023171     165.31692       -1.4022259   0.4236024   -3.310240   0.0009322   0.0776799      1.4022259
## ENSG00000072501    3694.76013       -0.5604815   0.1707989   -3.281529   0.0010325   0.0794200      0.5604815
## ENSG00000070018     119.89049       -1.0921227   0.3512409   -3.109327   0.0018751   0.1339388      1.0921227
## ENSG00000059377     131.98111        0.8405094   0.2744635    3.062372   0.0021959   0.1463935      0.8405094
## ENSG00000008277     377.43955       -0.6368732   0.2136506   -2.980910   0.0028739   0.1796208      0.6368732
## ENSG00000005059     479.12528        0.4225402   0.1492246    2.831571   0.0046320   0.2289281      0.4225402
## ENSG00000012963    1829.21224        0.2471040   0.0861859    2.867104   0.0041425   0.2289281      0.2471040
## ENSG00000038427     100.87217       -1.0743654   0.3810269   -2.819658   0.0048075   0.2289281      1.0743654
## ENSG00000068079    1035.17996        0.4075632   0.1415563    2.879160   0.0039874   0.2289281      0.4075632</code></pre>
</div>
<div id="interactive-shiny-app" class="section level3">
<h3 class="hasAnchor">
<a href="#interactive-shiny-app" class="anchor"></a>Interactive shiny-app</h3>
<p>Browsing gene expression can help to validate results or select some gene for downstream analysis. Run the following lines if you want to visualize your expression values by condition:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/degObj.html">degObj</a></span>(counts, design, <span class="st">"degObj.rda"</span>)
<span class="kw">library</span>(shiny)
shiny<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/shiny/topics/runUrl">runGitHub</a></span>(<span class="st">"lpantano/shiny"</span>, <span class="dt">subdir=</span><span class="st">"expression"</span>)</code></pre></div>
</div>
</div>
<div id="detect-patterns-of-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#detect-patterns-of-expression" class="anchor"></a>Detect patterns of expression</h2>
<p>In this section, we show how to detect pattern of expression. Mainly useful when data is a time course experiment. <code>degPatterns</code> needs a expression matrix, the design experiment and the column used to group samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ma =<span class="st"> </span><span class="kw">assay</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/rlog">rlog</a></span>(dds))[<span class="kw">row.names</span>(res)[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>],]
res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/degPatterns.html">degPatterns</a></span>(ma, design, <span class="dt">time =</span> <span class="st">"group"</span>)</code></pre></div>
<p><img src="DEGreport_files/figure-html/pattern-1.png" width="700"></p>
</div>
<div id="useful-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#useful-functions" class="anchor"></a>Useful functions</h2>
<p>This section shows some useful functions during DEG analysis.</p>
<div id="filter-genes-by-group" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-genes-by-group" class="anchor"></a>Filter genes by group</h3>
<p><code>degFilter</code> helps to filter genes with a minimum read count by group.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">cat</span>(<span class="st">"gene in original count matrix: 1000"</span>)</code></pre></div>
<p>gene in original count matrix: 1000</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter_count &lt;-<span class="st"> </span><span class="kw"><a href="../reference/degFilter.html">degFilter</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds),
                          design, <span class="st">"group"</span>,
                          <span class="dt">min=</span><span class="dv">1</span>, <span class="dt">minreads =</span> <span class="dv">50</span>)
<span class="kw">cat</span>(<span class="st">"gene in final count matrix"</span>, <span class="kw">nrow</span>(filter_count))</code></pre></div>
<p>gene in final count matrix 940</p>
</div>
<div id="generate-colors-for-metadata-variables" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-colors-for-metadata-variables" class="anchor"></a>Generate colors for metadata variables</h3>
<p>This functions allows you to create colors for metadata columns to be used as annotation for columns in a heatmap figure.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ComplexHeatmap)
th &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ComplexHeatmap/topics/HeatmapAnnotation">HeatmapAnnotation</a></span>(<span class="dt">df =</span> <span class="kw">colData</span>(dds),
                        <span class="dt">col =</span> <span class="kw"><a href="../reference/degColors.html">degColors</a></span>(<span class="kw">colData</span>(dds), <span class="ot">TRUE</span>))
<span class="kw"><a href="http://www.rdocumentation.org/packages/ComplexHeatmap/topics/Heatmap">Heatmap</a></span>(<span class="kw">log2</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds) <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,],
        <span class="dt">top_annotation =</span> th)</code></pre></div>
<p><img src="DEGreport_files/figure-html/degColors-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pheatmap)
<span class="kw"><a href="http://www.rdocumentation.org/packages/pheatmap/topics/pheatmap">pheatmap</a></span>(<span class="kw">log2</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds) <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,], 
         <span class="dt">annotation_col =</span> <span class="kw">as.data.frame</span>(<span class="kw">colData</span>(dds))[,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>],
         <span class="dt">annotation_colors =</span> <span class="kw"><a href="../reference/degColors.html">degColors</a></span>(<span class="kw">colData</span>(dds)[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>],
                                       <span class="dt">con_values =</span> <span class="kw">c</span>(<span class="st">"white"</span>,
                                                      <span class="st">"red"</span>)
                                       )
         )</code></pre></div>
<p><img src="DEGreport_files/figure-html/degColors-2.png" width="700"></p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.4 (2018-03-15)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.3
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] pheatmap_1.0.10            ComplexHeatmap_1.17.1     
##  [3] bindrcpp_0.2.2.9000        quantreg_5.36             
##  [5] SparseM_1.77               DESeq2_1.18.1             
##  [7] SummarizedExperiment_1.8.1 DelayedArray_0.4.1        
##  [9] matrixStats_0.53.1         Biobase_2.38.0            
## [11] GenomicRanges_1.30.3       GenomeInfoDb_1.14.0       
## [13] IRanges_2.12.0             S4Vectors_0.16.0          
## [15] BiocGenerics_0.24.0        DEGreport_1.17.3          
## [17] BiocStyle_2.6.1           
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.3-2            rjson_0.2.20               
##   [3] rprojroot_1.3-2             circlize_0.4.4             
##   [5] htmlTable_1.12              XVector_0.18.0             
##   [7] GlobalOptions_0.1.0         base64enc_0.1-3            
##   [9] ggdendro_0.1-20             fs_1.2.2                   
##  [11] rstudioapi_0.7              roxygen2_6.0.1.9000        
##  [13] MatrixModels_0.4-1          ggrepel_0.8.0              
##  [15] bit64_0.9-7                 AnnotationDbi_1.40.0       
##  [17] xml2_1.2.0                  splines_3.4.4              
##  [19] logging_0.7-103             mnormt_1.5-5               
##  [21] geneplotter_1.56.0          knitr_1.20                 
##  [23] Formula_1.2-3               Nozzle.R1_1.1-1            
##  [25] broom_0.4.4                 annotate_1.56.2            
##  [27] cluster_2.0.7-1             compiler_3.4.4             
##  [29] backports_1.1.2             assertthat_0.2.0           
##  [31] Matrix_1.2-14               lazyeval_0.2.1             
##  [33] cli_1.0.0                   limma_3.34.9               
##  [35] acepack_1.4.1               htmltools_0.3.6            
##  [37] tools_3.4.4                 gtable_0.2.0               
##  [39] glue_1.2.0                  GenomeInfoDbData_1.0.0     
##  [41] reshape2_1.4.3              dplyr_0.7.6                
##  [43] Rcpp_0.12.17                pkgdown_1.1.0              
##  [45] nlme_3.1-137                psych_1.8.4                
##  [47] stringr_1.3.1               XML_3.98-1.11              
##  [49] edgeR_3.20.9                zlibbioc_1.24.0            
##  [51] MASS_7.3-50                 scales_0.5.0               
##  [53] RColorBrewer_1.1-2          yaml_2.1.19                
##  [55] memoise_1.1.0               gridExtra_2.3              
##  [57] ggplot2_3.0.0               rpart_4.1-13               
##  [59] reshape_0.8.7               latticeExtra_0.6-28        
##  [61] stringi_1.2.3               RSQLite_2.1.1              
##  [63] highr_0.7                   genefilter_1.60.0          
##  [65] desc_1.2.0                  checkmate_1.8.5            
##  [67] BiocParallel_1.12.0         shape_1.4.4                
##  [69] rlang_0.2.1                 pkgconfig_2.0.1            
##  [71] commonmark_1.5              bitops_1.0-6               
##  [73] evaluate_0.10.1             lattice_0.20-35            
##  [75] purrr_0.2.5                 bindr_0.1.1.9000           
##  [77] labeling_0.3                htmlwidgets_1.2            
##  [79] cowplot_0.9.2               bit_1.1-14                 
##  [81] tidyselect_0.2.4            plyr_1.8.4                 
##  [83] magrittr_1.5                R6_2.2.2                   
##  [85] Hmisc_4.1-1                 DBI_1.0.0                  
##  [87] withr_2.1.2                 pillar_1.2.3               
##  [89] foreign_0.8-70              survival_2.42-3            
##  [91] RCurl_1.95-4.10             nnet_7.3-12                
##  [93] tibble_1.4.2                crayon_1.3.4               
##  [95] utf8_1.1.3                  rmarkdown_1.10             
##  [97] GetoptLong_0.1.7            locfit_1.5-9.1             
##  [99] data.table_1.11.4           blob_1.1.1                 
## [101] ConsensusClusterPlus_1.42.0 digest_0.6.15              
## [103] xtable_1.8-2                tidyr_0.8.1                
## [105] munsell_0.5.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#general-qc-figures-from-de-analysis">General QC figures from DE analysis</a></li>
      <li><a href="#report-from-deseq2-analysis">Report from DESeq2 analysis</a></li>
      <li><a href="#detect-patterns-of-expression">Detect patterns of expression</a></li>
      <li><a href="#useful-functions">Useful functions</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://lpantano.github.io">Lorena Pantano</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
