<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Make groups of genes using expression profile. — degPatterns • DEGreport</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Make groups of genes using expression profile. — degPatterns" />

<meta property="og:description" content="Note that this function doesn't calculate significant
difference between groups, so the
matrix used as input should be already filtered to contain only
genes that are significantly different or the most interesting genes
to study." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEGreport</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.17.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEGreport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Make groups of genes using expression profile.</h1>
    
    <div class="hidden name"><code>degPatterns.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Note that this function doesn't calculate significant
difference between groups, so the
matrix used as input should be already filtered to contain only
genes that are significantly different or the most interesting genes
to study.</p>
    
    </div>

    <pre class="usage"><span class='fu'>degPatterns</span>(<span class='no'>ma</span>, <span class='no'>metadata</span>, <span class='kw'>minc</span> <span class='kw'>=</span> <span class='fl'>15</span>, <span class='kw'>summarize</span> <span class='kw'>=</span> <span class='st'>"merge"</span>, <span class='kw'>time</span> <span class='kw'>=</span> <span class='st'>"time"</span>,
  <span class='kw'>col</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>consensusCluster</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>reduce</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fl'>0.7</span>,
  <span class='kw'>scale</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>pattern</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>groupDifference</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>eachStep</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>plot</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>fixy</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ma</th>
      <td><p>log2 normalized count matrix</p></td>
    </tr>
    <tr>
      <th>metadata</th>
      <td><p>data frame with sample information. Rownames
should match <code>ma</code> column names
row number should be the same length than p-values vector.</p></td>
    </tr>
    <tr>
      <th>minc</th>
      <td><p>integer minimum number of genes in a group that
will be return</p></td>
    </tr>
    <tr>
      <th>summarize</th>
      <td><p>character column name in metadata that will be used to group
replicates. If the column doesn't exist it'll merge the <code>time</code> and
the <code>col</code> columns, if <code>col</code> doesn't exist it'll use <code>time</code> only.
For instance, a merge between summarize and time parameters:
control_point0 ... etc</p></td>
    </tr>
    <tr>
      <th>time</th>
      <td><p>character column name in metadata that will be used as
variable that changes, normally a time variable.</p></td>
    </tr>
    <tr>
      <th>col</th>
      <td><p>character column name in metadata to separate
samples. Normally control/mutant</p></td>
    </tr>
    <tr>
      <th>consensusCluster</th>
      <td><p>Indicates whether using ConsensusClusterPlus
or <code><a href='http://www.rdocumentation.org/packages/cluster/topics/diana'>cluster::diana()</a></code></p></td>
    </tr>
    <tr>
      <th>reduce</th>
      <td><p>boolean reduce number of clusters using
correlation values between them.</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>integer threshold for correlation
expression to merge clusters (0 - 1)</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>boolean scale the <code>ma</code> values by row</p></td>
    </tr>
    <tr>
      <th>pattern</th>
      <td><p>numeric vector to be used to find patterns like this
from the count matrix. As well, it can be a character indicating the
genes inside the count matrix to be used as reference.</p></td>
    </tr>
    <tr>
      <th>groupDifference</th>
      <td><p>Minimum abundance difference between the
maximum value and minimum value for each feature. Please,
provide the value in the same range than the <code>ma</code> value
( if <code>ma</code> is in log2, <code>groupDifference</code> should be inside that range).</p></td>
    </tr>
    <tr>
      <th>eachStep</th>
      <td><p>Whether apply <code>groupDifference</code> at each stem over
<code>time</code> variable. <strong>This only work properly for one group
with multiple time points</strong>.</p></td>
    </tr>
    <tr>
      <th>plot</th>
      <td><p>boolean plot the clusters found</p></td>
    </tr>
    <tr>
      <th>fixy</th>
      <td><p>vector integers used as ylim in plot</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list wiht two items:</p><ul>
<li><p><code>df</code> is a data.frame
with two columns. The first one with genes, the second
with the clusters they belong.</p></li>
<li><p><code>pass</code> is a vector of the clusters that pass the <code>minc</code> cutoff.</p></li>
<li><p><code>plot</code> ggplot figure.</p></li>
<li><p><code>hr</code> clustering of the genes in hclust format.</p></li>
<li><p><code>profile</code> normalized count data used in the plot.</p></li>
<li><p><code>raw</code> data.frame with gene values summarized by biological replicates and
with metadata information attached.</p></li>
<li><p><code>summarise</code> data.frame with clusters values summarized by group and
with the metadata information attached.</p></li>
</ul>

    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>It can work with one or more groups with 2 or
more several time points.
Before calculating the genes similarity among samples,
all samples inside the same time point (<code>time</code> parameter) and
group (<code>col</code> paramater) are collapsed together, and the <code>mean</code>
value is the representation of the group for the gene abundance.
Then, all pair-wise gene expression are calculated using
<code>cor.test</code>  R function using kendall as the statistical
method. A distance matrix is created from those vsalues.
After that, <code><a href='http://www.rdocumentation.org/packages/cluster/topics/diana'>cluster::diana()</a></code> is used for the
clustering of gene-gene distance matrix and cut of the tree using
the divisive coefficient of the clustering, giving as well by diana.
Alernatively, if <code>consensusCluster</code> is on, it would used
ConsensusClusterPlus to cut the tree in stable clusters.
Finally, for each group of genes, only the ones that have genes
higher than <code>minc</code> parameter will be added to the figure.
The y-axis in the figure is the results of applying <code>scale()</code>
R function, what is similar to the meaning of creating a
<code>Z-score</code> where values are centered to the <code>mean</code>  and
scaled to the <code>standard desviation</code>.</p>
<p>The different patterns can be merged
to get similar ones into only one pattern. The expression
correlation of the patterns will be used to decide whether
some need to be merged or not.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>humanGender</span>)
<span class='fu'>library</span>(<span class='no'>SummarizedExperiment</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,</span>
#&gt; <span class='message'>    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,</span>
#&gt; <span class='message'>    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,</span>
#&gt; <span class='message'>    table, tapply, union, unique, unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DelayedArray</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘matrixStats’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘DelayedArray’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    apply</span></div><div class='input'><span class='fu'>library</span>(<span class='no'>ggplot2</span>)</div><div class='output co'>#&gt; <span class='message'>Want to understand how all the pieces fit together? See the R for Data</span>
#&gt; <span class='message'>Science book: http://r4ds.had.co.nz/</span></div><div class='input'><span class='no'>ma</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>assays</a></span>(<span class='no'>humanGender</span>)<span class='kw'>[[</span><span class='fl'>1</span>]][<span class='fl'>1</span>:<span class='fl'>100</span>,]
<span class='no'>des</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class'>colData</a></span>(<span class='no'>humanGender</span>)
<span class='no'>des</span><span class='kw'>[[</span><span class='st'>"other"</span>]] <span class='kw'>&lt;-</span> <span class='fu'>sample</span>(<span class='fu'>c</span>(<span class='st'>"a"</span>, <span class='st'>"b"</span>), <span class='fl'>85</span>, <span class='kw'>replace</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>degPatterns</span>(<span class='no'>ma</span>, <span class='no'>des</span>, <span class='kw'>time</span><span class='kw'>=</span><span class='st'>"group"</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='st'>"other"</span>)</div><div class='output co'>#&gt; <span class='message'>Working with 100 genes.</span></div><div class='output co'>#&gt; <span class='message'>Working with 66 genes after filtering: minc &gt; 15</span></div><div class='output co'>#&gt; <span class='message'>Joining, by = "merge"</span></div><div class='output co'>#&gt; <span class='message'>Joining, by = "merge"</span></div><div class='img'><img src='degPatterns-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'># Use the data yourself for custom figures</span>
 <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/ggplot'>ggplot</a></span>(<span class='no'>res</span><span class='kw'>[[</span><span class='st'>"normalized"</span>]],
        <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/aes'>aes</a></span>(<span class='no'>group</span>, <span class='no'>value</span>, <span class='kw'>color</span> <span class='kw'>=</span> <span class='no'>other</span>, <span class='kw'>fill</span> <span class='kw'>=</span> <span class='no'>other</span>)) +
  <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/geom_boxplot'>geom_boxplot</a></span>() +
   <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/geom_point'>geom_point</a></span>(<span class='kw'>position</span> <span class='kw'>=</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/position_jitterdodge'>position_jitterdodge</a></span>(<span class='kw'>dodge.width</span> <span class='kw'>=</span> <span class='fl'>0.9</span>)) +
   <span class='co'># change the method to make it smoother</span>
   <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/geom_smooth'>geom_smooth</a></span>(<span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/aes'>aes</a></span>(<span class='kw'>group</span><span class='kw'>=</span><span class='no'>other</span>), <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"lm"</span>)</div><div class='img'><img src='degPatterns-2.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://lpantano.github.io'>Lorena Pantano</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

