<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>DEGSet — DEGSet • DEGreport</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="DEGSet — DEGSet" />

<meta property="og:description" content="S4 class to store data from differentially expression analysis.
It should be compatible with different package and stores the information
in a way the methods will work with all of them." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEGreport</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.17.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DEGreport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DEGSet</h1>
    
    <div class="hidden name"><code>DEGSet.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>S4 class to store data from differentially expression analysis.
It should be compatible with different package and stores the information
in a way the methods will work with all of them.</p>
    
    </div>

    <pre class="usage"><span class='fu'>DEGSet</span>(<span class='no'>resList</span>, <span class='no'>default</span>)

<span class='fu'>DEGSet</span>(<span class='no'>resList</span>, <span class='no'>default</span>)

<span class='fu'>as.DEGSet</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S4 method for TopTags</span>
<span class='fu'>as.DEGSet</span>(<span class='no'>object</span>, <span class='kw'>default</span> <span class='kw'>=</span> <span class='st'>"raw"</span>, <span class='kw'>extras</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for data.frame</span>
<span class='fu'>as.DEGSet</span>(<span class='no'>object</span>, <span class='no'>contrast</span>, <span class='kw'>default</span> <span class='kw'>=</span> <span class='st'>"raw"</span>,
  <span class='kw'>extras</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for DESeqResults</span>
<span class='fu'>as.DEGSet</span>(<span class='no'>object</span>, <span class='kw'>default</span> <span class='kw'>=</span> <span class='st'>"shrunken"</span>,
  <span class='kw'>extras</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>resList</th>
      <td><p>List with results as elements containing log2FoldChange,
pvalues and padj as column. Rownames should be feature names. Elements
should have names.</p></td>
    </tr>
    <tr>
      <th>default</th>
      <td><p>The name of the element to use by default.</p></td>
    </tr>
    <tr>
      <th>object</th>
      <td><p>Different objects to be transformed to DEGSet when using <code>as.DEGSet</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Optional parameters of the generic.</p></td>
    </tr>
    <tr>
      <th>extras</th>
      <td><p>List of extra tables related to the same comparison when using <code>as.DEGSet</code>.</p></td>
    </tr>
    <tr>
      <th>contrast</th>
      <td><p>To name the comparison when using <code>as.DEGSet</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>For now supporting only <code><a href='http://www.rdocumentation.org/packages/DESeq2/topics/results'>DESeq2::results()</a></code> output.
Use constructor <code><a href='degComps.html'>degComps()</a></code> to create the object.</p>
<p>The list will contain one element for each comparison done.
Each element has the following structure:</p><ul>
<li><p>DEG table</p></li>
<li><p>Optional table with shrunk Fold Change when it has been done.</p></li>
</ul>
    <p>To access the raw table use <code><a href='deg.html'>deg(dgs, "raw")</a></code>, to access the
shrunken table use <code><a href='deg.html'>deg(dgs, "shrunken")</a></code> or just <code><a href='deg.html'>deg(dgs)</a></code>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>library</span>(<span class='no'>DESeq2</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,</span>
#&gt; <span class='message'>    colnames, colSums, dirname, do.call, duplicated, eval, evalq,</span>
#&gt; <span class='message'>    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,</span>
#&gt; <span class='message'>    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,</span>
#&gt; <span class='message'>    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,</span>
#&gt; <span class='message'>    rowSums, sapply, setdiff, sort, table, tapply, union, unique,</span>
#&gt; <span class='message'>    unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: SummarizedExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DelayedArray</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘matrixStats’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocParallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘DelayedArray’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    aperm, apply</span></div><div class='input'><span class='fu'>library</span>(<span class='no'>edgeR</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: limma</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘limma’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:DESeq2’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    plotMA</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:BiocGenerics’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    plotMA</span></div><div class='input'><span class='fu'>library</span>(<span class='no'>limma</span>)
<span class='no'>dds</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/DESeq2/topics/makeExampleDESeqDataSet'>makeExampleDESeqDataSet</a></span>(<span class='kw'>betaSD</span> <span class='kw'>=</span> <span class='fl'>1</span>)
<span class='fu'>colData</span>(<span class='no'>dds</span>)<span class='kw'>[[</span><span class='st'>"treatment"</span>]] <span class='kw'>&lt;-</span> <span class='fu'>sample</span>(<span class='fu'>colData</span>(<span class='no'>dds</span>)<span class='kw'>[[</span><span class='st'>"condition"</span>]], <span class='fl'>12</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/DESeq2/topics/design'>design</a></span>(<span class='no'>dds</span>) <span class='kw'>&lt;-</span>  ~ <span class='no'>condition</span> + <span class='no'>treatment</span>
<span class='no'>dds</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/DESeq2/topics/DESeq'>DESeq</a></span>(<span class='no'>dds</span>)</div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='input'><span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='degComps.html'>degComps</a></span>(<span class='no'>dds</span>, <span class='kw'>combs</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"condition"</span>))</div><div class='output co'>#&gt; <span class='message'>Doing 1 element(s).</span></div><div class='output co'>#&gt; <span class='message'>Doing results() for each element.</span></div><div class='output co'>#&gt; <span class='message'>Doing lcfSrink() for each element.</span></div><div class='input'><span class='fu'><a href='deg.html'>deg</a></span>(<span class='no'>res</span><span class='kw'>[[</span><span class='fl'>1</span>]])</div><div class='output co'>#&gt; log2 fold change (MAP): condition B vs A 
#&gt; Wald test p-value: condition B vs A 
#&gt; DataFrame with 1000 rows and 6 columns
#&gt;                  baseMean     log2FoldChange             lfcSE
#&gt;                 &lt;numeric&gt;          &lt;numeric&gt;         &lt;numeric&gt;
#&gt; gene202  595.738080502665   2.69290441661694 0.312772825764591
#&gt; gene710  117.423055944441   2.39202225442696 0.309387809822349
#&gt; gene365  62.4434899710059   2.68819163853763 0.347014786363427
#&gt; gene364  504.522116500683   2.13726876442053 0.304160581713146
#&gt; gene92   220.780204589055   2.05762907387256 0.326874656707121
#&gt; ...                   ...                ...               ...
#&gt; gene984  3.56063169601722   1.18342031931477 0.558212799249338
#&gt; gene986  1.03912504596143 -0.131696247712928 0.550000503156253
#&gt; gene989  1.84138670225673  0.797083476699305 0.567712937423255
#&gt; gene995  2.88636057994642 0.0840813792066131 0.555663728011982
#&gt; gene1000  2.5816660463203 -0.535343293319606 0.564961445050577
#&gt;                        stat               pvalue                 padj
#&gt;                   &lt;numeric&gt;            &lt;numeric&gt;            &lt;numeric&gt;
#&gt; gene202    7.82879005947595 4.92587384985754e-15 4.05892005228261e-12
#&gt; gene710    6.93369964061162  4.0997464674756e-12 1.68909554459995e-09
#&gt; gene365    6.79494555755484 1.08353438828276e-11  2.9761077864833e-09
#&gt; gene364    6.56399761634188 5.23840391930272e-11 1.07911120737636e-08
#&gt; gene92      5.4297680851086 5.64273306072517e-08 9.29922408407507e-06
#&gt; ...                     ...                  ...                  ...
#&gt; gene984    1.56606943948116    0.117332361646146                   NA
#&gt; gene986  -0.743786405055134    0.457005706109876                   NA
#&gt; gene989    1.31499465077388    0.188511700254376                   NA
#&gt; gene995    1.09138089416803    0.275105315817471                   NA
#&gt; gene1000 -0.806243807885651    0.420102283688199                   NA</div><div class='input'><span class='fu'><a href='deg.html'>deg</a></span>(<span class='no'>res</span><span class='kw'>[[</span><span class='fl'>1</span>]], <span class='kw'>tidy</span> <span class='kw'>=</span> <span class='st'>"tibble"</span>)</div><div class='output co'>#&gt; # A tibble: 1,000 x 7
#&gt;    gene    baseMean log2FoldChange lfcSE  stat   pvalue     padj
#&gt;  * &lt;chr&gt;      &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
#&gt;  1 gene202    596.            2.69 0.313  7.83 4.93e-15 4.06e-12
#&gt;  2 gene710    117.            2.39 0.309  6.93 4.10e-12 1.69e- 9
#&gt;  3 gene365     62.4           2.69 0.347  6.79 1.08e-11 2.98e- 9
#&gt;  4 gene364    505.            2.14 0.304  6.56 5.24e-11 1.08e- 8
#&gt;  5 gene92     221.            2.06 0.327  5.43 5.64e- 8 9.30e- 6
#&gt;  6 gene664    110.            1.97 0.350  5.20 2.03e- 7 2.79e- 5
#&gt;  7 gene577    135.           -1.66 0.295 -5.17 2.37e- 7 2.79e- 5
#&gt;  8 gene564     83.3          -1.93 0.361 -5.12 3.08e- 7 3.10e- 5
#&gt;  9 gene787     71.0           1.92 0.360  5.08 3.76e- 7 3.10e- 5
#&gt; 10 gene836     49.2           2.25 0.377  5.08 3.71e- 7 3.10e- 5
#&gt; # ... with 990 more rows</div><div class='input'><span class='co'># From edgeR</span>
<span class='no'>dge</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/edgeR/topics/DGEList'>DGEList</a></span>(<span class='kw'>counts</span><span class='kw'>=</span><span class='fu'><a href='http://www.rdocumentation.org/packages/DESeq2/topics/counts'>counts</a></span>(<span class='no'>dds</span>), <span class='kw'>group</span><span class='kw'>=</span><span class='fu'>colData</span>(<span class='no'>dds</span>)<span class='kw'>[[</span><span class='st'>"treatment"</span>]])
<span class='no'>dge</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/edgeR/topics/estimateCommonDisp'>estimateCommonDisp</a></span>(<span class='no'>dge</span>)
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>as.DEGSet</span>(<span class='fu'><a href='http://www.rdocumentation.org/packages/edgeR/topics/topTags'>topTags</a></span>(<span class='fu'><a href='http://www.rdocumentation.org/packages/edgeR/topics/exactTest'>exactTest</a></span>(<span class='no'>dge</span>)))
<span class='co'># From limma</span>
<span class='no'>v</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/limma/topics/voom'>voom</a></span>(<span class='fu'><a href='http://www.rdocumentation.org/packages/DESeq2/topics/counts'>counts</a></span>(<span class='no'>dds</span>), <span class='fu'>model.matrix</span>(~<span class='no'>treatment</span>, <span class='fu'>colData</span>(<span class='no'>dds</span>)), <span class='kw'>plot</span><span class='kw'>=</span><span class='fl'>FALSE</span>)
<span class='no'>fit</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/limma/topics/lmFit'>lmFit</a></span>(<span class='no'>v</span>)
<span class='no'>fit</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/limma/topics/ebayes'>eBayes</a></span>(<span class='no'>fit</span>, <span class='kw'>robust</span><span class='kw'>=</span><span class='fl'>TRUE</span>)
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>as.DEGSet</span>(<span class='fu'><a href='http://www.rdocumentation.org/packages/limma/topics/toptable'>topTable</a></span>(<span class='no'>fit</span>, <span class='kw'>n</span> <span class='kw'>=</span> <span class='st'>"Inf"</span>), <span class='st'>"A_vs_B"</span>)</div><div class='output co'>#&gt; <span class='message'>Removing intercept from test coefficients</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Lorena Pantano

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://lpantano.github.io'>Lorena Pantano</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

